{
    "sourceFile": "src/services/securityDashboardService.js",
    "activeCommit": 0,
    "commits": [
        {
            "activePatchIndex": 1,
            "patches": [
                {
                    "date": 1753420450423,
                    "content": "Index: \n===================================================================\n--- \n+++ \n"
                },
                {
                    "date": 1753420529988,
                    "content": "Index: \n===================================================================\n--- \n+++ \n@@ -0,0 +1,57 @@\n+import api from \"./api\";\r\n+import { API_ENDPOINTS } from \"../utils/constants\";\r\n+\r\n+export const createEmergency = async (emergencyData) => {\r\n+    try{\r\n+        const url = API_ENDPOINTS.CREATE_EMERGENCY;\r\n+        const token = localStorage.getItem(\"token\");\r\n+        const response = await api.post(url, emergencyData, {\r\n+            headers: {\r\n+                'Content-Type': 'application/json',\r\n+                'Authorization': `Bearer ${token}`\r\n+            }\r\n+        });\r\n+        const result = response.data;\r\n+        if(result.result_code === 0){\r\n+            return result.data;\r\n+        }else{\r\n+            throw new Error(result.message || \"Emergency creation failed.\");\r\n+        }\r\n+    } catch(error){\r\n+        throw error.message || \"Something went wrong while creating emergencies.\";\r\n+    } finally{\r\n+        console.log(\"Emergency created successfully\");\r\n+    }\r\n+};\r\n+\r\n+export const createIncidence = async (incidenceData) => {\r\n+    try{\r\n+        const url = API_ENDPOINTS.CREATE_INCIDENCE;\r\n+        const response = await api.post(url, incidenceData);\r\n+        const result = response.data;\r\n+        if(result.result_code === 0){\r\n+            return result.data;\r\n+        }else{\r\n+            throw new Error(result.message || \"Incidence creation failed.\");\r\n+        }\r\n+    } catch(error){\r\n+        throw error.message || \"Something went wrong while creating the incidence.\";\r\n+    } finally {\r\n+        console.log(\"Incidence created successfully\");\r\n+    }\r\n+};\r\n+\r\n+export const fetchStatistics = async () =>{\r\n+    try{\r\n+        const response = await api.get(API_ENDPOINTS.STATISTICS);\r\n+        const result = response.data;\r\n+        if(result.result_code === 0){\r\n+            return result.data;\r\n+        }else{\r\n+            throw new Error(result.message || \"Failed to fetch statistics.\");\r\n+        }\r\n+    } catch(error){\r\n+        throw error.message || \"Something went wrong while fetching statistics.\";\r\n+    }\r\n+};\r\n+\r\n"
                }
            ],
            "date": 1753420450423,
            "name": "Commit-0",
            "content": "import api from \"./api\";\r\nimport { API_ENDPOINTS } from \"../utils/constants\";\r\n\r\nexport const createEmergency = async (emergencyData) => {\r\n    try{\r\n        const url = API_ENDPOINTS.CREATE_EMERGENCY;\r\n        const token = localStorage.getItem(\"token\");\r\n        const response = await api.post(url, emergencyData, {\r\n            headers: {\r\n                'Content-Type': 'application/json',\r\n                'Authorization': `Bearer ${token}`\r\n            }\r\n        });\r\n        const result = response.data;\r\n        if(result.result_code === 0){\r\n            return result.data;\r\n        }else{\r\n            throw new Error(result.message || \"Emergency creation failed.\");\r\n        }\r\n    } catch(error){\r\n        throw error.message || \"Something went wrong while creating emergencies.\";\r\n    } finally{\r\n        console.log(\"Emergency created successfully\");\r\n    }\r\n};\r\n\r\nexport const createIncidence = async (incidenceData) => {\r\n    try{\r\n        const url = API_ENDPOINTS.CREATE_INCIDENCE;\r\n        const response = await api.post(url, incidenceData);\r\n        const result = response.data;\r\n        if(result.result_code === 0){\r\n            return result.data;\r\n        }else{\r\n            throw new Error(result.message || \"Incidence creation failed.\");\r\n        }\r\n    } catch(error){\r\n        throw error.message || \"Something went wrong while creating the incidence.\";\r\n    } finally {\r\n        console.log(\"Incidence created successfully\");\r\n    }\r\n};\r\n\r\nexport const fetchStatistics = async () =>{\r\n    try{\r\n        const response = await api.get(API_ENDPOINTS.STATISTICS);\r\n        const result = response.data;\r\n        if(result.result_code === 0){\r\n            return result.data;\r\n        }else{\r\n            throw new Error(result.message || \"Failed to fetch statistics.\");\r\n        }\r\n    } catch(error){\r\n        throw error.message || \"Something went wrong while fetching statistics.\";\r\n    }\r\n};\r\n\r\n"
        }
    ]
}